### PR TITLE
dotCMS/core#25315 [UI] Convert Start Experiment to 3 dots menu (In Experiment Configuration)

### LINKED ISSUE #25315: [UI] Convert Start Experiment to 3 dots menu (In Experiment Configuration)
### Parent Issue

N/A

### User Story

Replace the "Start Experiment Button" with the 3 dot menu. 

### Acceptance Criteria

3 dot menu should be in the top right of the config page, with only the Start Experiment option. 

image

https://www.figma.com/file/fAoCk4swhiwplkU5GP21YD/Portlet-%2F-Experiments?type=design&node-id=3706-28526&mode=design&t=Mpy6ZK5RyIiGATfU-0

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Add 3 dot menu in the top right of experiment configuration. 

#### Menu 

image

#### Open 

image

#### Confirmations

image

image

### CODE REVIEW

**Review by fmontes — DISMISSED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/store/dot-experiments-configuration-store.ts [hunk 1]
```diff
@@ -879,4 +877,70 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
             ]
         };
     }
+
+    private disableStartExperiment(experiment: DotExperiment): boolean {
+        return experiment?.trafficProportion.variants.length < 2 || !experiment?.goals;
+    }
+
+    private setMenuItems(experiment: DotExperiment): MenuItem[] {
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/store/dot-experiments-configuration-store.ts]: done

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/dot-experiments-configuration.component.ts [hunk 2]
```diff
@@ -47,7 +55,9 @@ import { DotExperimentsInlineEditTextComponent } from '../shared/ui/dot-experime
         RippleModule,
         InplaceModule,
         InputTextModule,
-        RippleModule
+        RippleModule,
```
[the remark this answers is not in the document]
Comment (hmoreras) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/dot-experiments-configuration.component.ts]: removed

File: core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts [hunk 3]
```diff
@@ -209,3 +209,5 @@ export const ReportSummaryLegendByBayesianStatus: Record<BayesianLegendStatus, S
         legend: 'experiments.summary.suggested-winner.no-enough-sessions'
     }
 };
+
+export const CONFIRM_DIALOG_KEY = 'confirmDialog';
```
Comment (oidacra) [inline core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts:213]: This is a unique key only for Configuration's Dialog? Could be something like, CONFIGURATION_CONFIRM_DIALOG_KEY

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-targeting/dot-experiments-configuration-targeting.component.spec.ts [hunk 4]
```diff
@@ -52,7 +52,8 @@ describe('DotExperimentsConfigurationTargetingComponent', () => {
             {
                 provide: DotMessageService,
                 useValue: messageServiceMock
-            }
+            },
+            mockProvider(ConfirmationService)
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-targeting/dot-experiments-configuration-targeting.component.spec.ts:56]: It is necessary this Provider in this component?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts [hunk 5]
```diff
@@ -58,7 +58,8 @@ describe('DotExperimentsConfigurationSchedulingAddComponent', () => {
             {
                 provide: DotMessageService,
                 useValue: messageServiceMock
-            }
+            },
+            mockProvider(ConfirmationService)
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-scheduling-add/dot-experiments-configuration-scheduling-add.component.spec.ts:62]: It is necessary this Provider in this component?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts [hunk 6]
```diff
@@ -70,6 +70,7 @@ describe('DotExperimentsConfigurationGoalSelectComponent', () => {
                 useValue: messageServiceMock
             },
             mockProvider(DotHttpErrorManagerService),
+            mockProvider(ConfirmationService),
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.spec.ts:73]: It is necessary this Provider in this component?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-allocation-add/dot-experiments-configuration-traffic-allocation-add.component.spec.ts [hunk 7]
```diff
@@ -52,6 +52,7 @@ describe('DotExperimentsConfigurationTrafficAllocationAddComponent', () => {
             mockProvider(MessageService),
             mockProvider(ActivatedRoute, ACTIVE_ROUTE_MOCK_CONFIG),
             mockProvider(DotHttpErrorManagerService),
+            mockProvider(ConfirmationService),
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-allocation-add/dot-experiments-configuration-traffic-allocation-add.component.spec.ts:55]: It is necessary this Provider in this component?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts [hunk 8]
```diff
@@ -62,7 +62,8 @@ describe('DotExperimentsConfigurationTrafficSplitAddComponent', () => {
             {
                 provide: DotMessageService,
                 useValue: messageServiceMock
-            }
+            },
+            mockProvider(ConfirmationService)
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-traffic-split-add/dot-experiments-configuration-traffic-split-add.component.spec.ts:66]: It is necessary this Provider in this component?

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-variants-add/dot-experiments-configuration-variants-add.component.spec.ts [hunk 9]
```diff
@@ -63,7 +63,8 @@ describe('DotExperimentsConfigurationVariantsAddComponent', () => {
                 provide: DotMessageService,
                 useValue: messageServiceMock
             },
-            mockProvider(MessageService)
+            mockProvider(MessageService),
+            mockProvider(ConfirmationService)
```
Comment (oidacra) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-configuration/components/dot-experiments-configuration-variants-add/dot-experiments-configuration-variants-add.component.spec.ts:67]: It is necessary this Provider in this component?

**Review by oidacra — APPROVED** [review]

File: core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/libs/dotcms-models/src/lib/dot-experiments-constants.ts:213]: yea just configuration.. will make that change

**Review by rjvelazco — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

**Pull request merged**